server/sonar-server/src/main/java/org/sonar/server/usertoken/ws/TokenPermissionsValidator.java

```diff
@@ -31,11 +30,8 @@ private TokenPermissionsValidator() {
   }
 
   static void validate(UserSession userSession, @Nullable String requestLogin) {
-    if (userSession.hasPermission(GlobalPermissions.SYSTEM_ADMIN)
-      || (requestLogin != null && requestLogin.equals(userSession.getLogin()))) {
-      return;
+    if (!userSession.isRoot() && !(requestLogin != null && requestLogin.equals(userSession.getLogin()))) {
```

Comment:
hard to read with all the `!`
what about applying boolean logic or just to separate if conditions following one another?